### PR TITLE
Conversion

### PR DESCRIPTION
**Pull request by cmontecoding**

### CODE REVIEW

File: src/Conversion.sol [hunk 1]
```diff
@@ -1,55 +1,150 @@
...
+
+    /// @param _kwenta $KWENTA token address on OE
+    /// @param _snx $SNX token address
+    /// @param _vestingStartTime Global start time for vesting
+    /// @param _snxInflationAmount Amount of SNX inflation deposited to the contract
+    /// @param _owner Owner of the contract
+    constructor(
+        address _kwenta,
+        address _snx,
+        uint256 _vestingStartTime,
```
Comment (jcmonte) [inline src/Conversion.sol:67]: Let's just make this a fixed constant -- November 15th 00:00 UTC -- formal acq. date.

File: src/Conversion.sol [hunk 2]
```diff
@@ -1,55 +1,150 @@
...
+        view
+        returns (uint256 vestableRemainder)
+    {
+        uint256 timeCliffEnds = vestingStartTime + VESTING_CLIFF_DURATION;
+        if (block.timestamp < timeCliffEnds) {
+            return 0;
+        }
+        if (block.timestamp > vestingStartTime + WITHDRAW_START) {
+            return 0;
+        }
```
Comment (jcmonte) [inline src/Conversion.sol:104]: Don't worry about this edge case. If all SNX is withdrawn, conversion is implicitly stopped. You can write a test for this.

File: src/Conversion.sol [hunk 3]
```diff
@@ -1,55 +1,150 @@
...
 
-    function vest(address to) public returns (uint256) {
-        /// get vestableAmount()
-        /// increase claimed balance
-        /// transfer SNX to
-        /// emit SNXVested(msg.sender, to, amount) event
+    /// @inheritdoc IConversion
+    function vest(address to) public returns (uint256 amountVested) {
+        amountVested = vestableAmount(msg.sender);
+        claimedSNX[msg.sender] += amountVested;
```
Comment (jcmonte) [inline src/Conversion.sol:138]: make a variable `caller = msg.sender` just for clarity

**Review by jcmonte — APPROVED** [review]

File: src/Conversion.sol [hunk 4]
```diff
@@ -1,55 +1,133 @@
...
+    uint256 public constant LINEAR_VESTING_DURATION = 270 days;
+
+    /// @notice Withdrawal start time in seconds (2 years)
+    uint256 public constant WITHDRAW_START = 730 days;
 
     /// @notice Global start time for vesting
+    /// @notice Friday, November 15, 2024 12:00:00 AM (GMT)
     /// @dev From this derive 3 months cliff 9 month linear vesting
-    uint256 public vestingStartTime;
+    uint256 public constant VESTING_START_TIME = 1_731_628_800;
```
Comment (jcmonte) [inline src/Conversion.sol:32]: underscores on timestamp is crazy
